### ISSUE TITLE
If a query fails, akka-workers goes into an infinite loop

### ISSUE THREAD

**Issue by z0u** [vpac-innovations/rsa#31]
If you try to run a query that has an error in it, rsaworker tries over and over again to run it. This would probably prevent any other queries from being run.

Example output (includes query XML):

```
[ERROR] [08/06/2014 15:38:38.398] [Workers-akka.actor.default-dispatcher-2] [akka://Workers/user/worker/exec] null
java.lang.NullPointerException
    at org.vpac.ndg.query.QueryDefinitionPreprocessor.setQueryDefinition(QueryDefinitionPreprocessor.java:95)
    at org.vpac.ndg.query.Query.setMemento(Query.java:140)
    at org.vpac.worker.WorkExecutor.executeQuery(WorkExecutor.java:183)
    at org.vpac.worker.WorkExecutor.onReceive(WorkExecutor.java:100)
    at akka.actor.UntypedActor$$anonfun$receive$1.applyOrElse(UntypedActor.scala:167)
    at akka.actor.Actor$class.aroundReceive(Actor.scala:465)
    at akka.actor.UntypedActor.aroundReceive(UntypedActor.scala:97)
    at akka.actor.ActorCell.receiveMessage(ActorCell.scala:516)
    at akka.actor.ActorCell.invoke(ActorCell.scala:487)
    at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:238)
    at akka.dispatch.Mailbox.run(Mailbox.scala:220)
    at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
    at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
    at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
    at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
    at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)

[INFO] [08/06/2014 15:38:38.400] [Workers-akka.actor.default-dispatcher-24] [akka://Workers/user/master/active] Work failed: Work{workId='480cbafd-68d5-46d1-81e7-a4979fd8b379', qd=<query>
  <input id="Rainfall_c" href="rsa:Rainfall/100m"/>
  <output id="output">
    <grid ref="#Rainfall_c"/>
    <variable name="mask" ref="#FOO/output"/>
  </output>
</query>, ver=netcdf4_classic, job=ff80818147a9d2d40147a9d37df30001, bound=Box((2262000.0, 2126800.0) - (2400000.0, 2500000.0)), outputResolution=m100}
```

**Comment by forjin-vpac-innovations** [comment]
That's true. 

I think I should have error handling process.

I'll deal with it.

**Comment by forjin-vpac-innovations** [comment]
solved on 6ad546c2f104fb078bc74a13816478a30931c835
